src/Microsoft.AspNetCore.Razor.Language/RazorConfiguration.cs

```diff
@@ -2,24 +2,58 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace Microsoft.AspNetCore.Razor.Language
 {
     public sealed class RazorConfiguration
     {
-        public static readonly RazorConfiguration Default = new RazorConfiguration(RazorLanguageVersion.Latest, designTime: false);
+        public static readonly RazorConfiguration Default = new RazorConfiguration(
+            RazorLanguageVersion.Latest, 
+            "unnamed",
+            Array.Empty<RazorExtension>(),
+            designTime: false);
 
-        public RazorConfiguration(RazorLanguageVersion languageVersion, bool designTime)
+        // This is used only in some back-compat scenarios. We don't expose it because there's no
+        // use case for anyone else to use it.
+        internal static readonly RazorConfiguration DefaultDesignTime = new RazorConfiguration(
+            RazorLanguageVersion.Latest,
+            "unnamed",
+            Array.Empty<RazorExtension>(),
+            designTime: true);
```

Comment:
@NTaylorMullen - design time being a part of the configuration is making me sad.